src/uct/ib/rdmacm/rdmacm_ep.c

```diff
@@ -78,3 +117,8 @@ UCS_CLASS_DEFINE_NEW_FUNC(uct_rdmacm_ep_t, uct_ep_t, uct_iface_t*,
                           uct_sockaddr_conn_reply_callback_t ,
                           void *, uint32_t, const void *, size_t);
 UCS_CLASS_DEFINE_DELETE_FUNC(uct_rdmacm_ep_t, uct_ep_t);
+
+void uct_rdmacm_ep_set_failed(uct_iface_t *iface, uct_ep_h ep, ucs_status_t status)
+{
+    uct_set_ep_failed(&UCS_CLASS_NAME(uct_rdmacm_ep_t), ep, iface, status);
```

Comment:
@evgeny-leksikov The iface->ep is accessed from the main thread, on creation and on destruction, where it's protected, and the rest of the accesses to it are from the async thread.
What reason to pretect do you see?

Reply:
if i understand correctly it can be called from async thread during creation. if you are sure ep can't be accessed this time from main thread, then keep it as is.

Reply:
creation of the ep (uct_ep_create_sockaddr()) is done only from the main thread - by the user.
the other time it is accesses from the main thread is during destroy, and there it is protected.